sys/src/go/cmd/tmpfs/tmpfs_test.go

```diff
@@ -0,0 +1,235 @@
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"log"
+	"net"
+	"testing"
+
+	"harvey-os.org/internal/tmpfs"
+	"harvey-os.org/pkg/ninep/protocol"
+)
+
+// Lots of tests shamelessly lifted from the ninep/ufs code
+
+func print(f string, args ...interface{}) {
+	fmt.Printf(f+"\n", args...)
+}
+
+// Create and add some files to the archive.
+func createTestImage() *bytes.Buffer {
+	var buf bytes.Buffer
+
+	gztw := gzip.NewWriter(&buf)
+	defer func() {
+		if err := gztw.Close(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	tw := tar.NewWriter(gztw)
+	defer func() {
+		if err := tw.Close(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	var files = []struct {
+		Name, Body string
+	}{
+		{"readme.txt", "This archive contains some text files."},
+		{"foo/gopher.txt", "Gopher names:\nGeorge\nGeoffrey\nGonzo"},
+		{"bar/todo.txt", "Get animal handling license."},
+		{"foo/todo2.txt", "harvey lalal"},
+		{"abc/123/sean.txt", "lorem ipshum."},
+		{"foo/bar/hello.txt", "hello world!"},
+	}
+	for _, file := range files {
+		hdr := &tar.Header{
+			Name: file.Name,
+			Mode: 0600,
+			Size: int64(len(file.Body)),
+		}
+		if err := tw.WriteHeader(hdr); err != nil {
+			log.Fatal(err)
+		}
+		if _, err := tw.Write([]byte(file.Body)); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	return &buf
+}
+
+func TestNwTmpfs(t *testing.T) {
+	fs, err := newTmpfs(nil)
+	if fs != nil || err == nil {
+		t.Fatal("no archive - should have failed")
+	}
+
+	arch := tmpfs.ReadImage(createTestImage())
+
+	fs, err = newTmpfs(arch)
+	if fs == nil || err != nil {
+		t.Fatal("should not have failed")
+	}
+}
+
+func TestMount(t *testing.T) {
```

Comment:
TODO: use iostats to trace a session on this, and from that, create a table driven test. 

Someday.